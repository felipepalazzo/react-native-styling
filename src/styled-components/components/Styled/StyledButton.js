import React from 'react'
import { Text } from 'react-native'
import styled from 'styled-components'
import { StyledDefaultButton } from './StyledDefaultButton'
import { COLORS } from '../../../variables'

const ButtonContainer = StyledDefaultButton.extend`
  background-color: ${props => props.primary ? COLORS.white : COLORS.blue};
  width: 60px;
`

const Label = styled(Text)`
  text-align: center;
`

const StyledButton = props =>
  <ButtonContainer onPress={props.onPress} primary={props.primary}>
    <Label>{props.children}</Label>
  </ButtonContainer>

export default StyledButton
