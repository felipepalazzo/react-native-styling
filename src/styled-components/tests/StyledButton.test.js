/* globals it, expect, describe */
import React from 'react'
import StyledButton from '../components/Styled/StyledButton'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('StyledButton', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <StyledButton secondary>
        Add
      </StyledButton>
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
