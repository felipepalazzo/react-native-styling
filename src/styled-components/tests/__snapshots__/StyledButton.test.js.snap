// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton renders correctly 1`] = `
.c0 {
  text-align: center;
}

<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="c0"
    ellipsizeMode="tail"
  >
    Add
  </Text>
</View>
`;
